packages/kbn-config/src/env.ts

```diff
@@ -17,12 +17,21 @@
  * under the License.
  */
 
-import { resolve, dirname } from 'path';
+import { resolve, join } from 'path';
+import loadJsonFile from 'load-json-file';
+import { REPO_ROOT } from '@kbn/dev-utils';
 import { PackageInfo, EnvironmentMode } from './types';
 
-// `require` is necessary for this to work inside x-pack code as well
-// eslint-disable-next-line @typescript-eslint/no-var-requires
-const pkg = require('../../../../package.json');
+// some tests are mocking the `fs` package while importing code from the `kbn/config` package,
+// causing the package read to fail. We use this memoize accessor trick to only read the package
+// at first actual usage.
+let kibanaPackageFile: any;
+const getRepositoryPackage = () => {
+  if (!kibanaPackageFile) {
+    kibanaPackageFile = loadJsonFile.sync(join(REPO_ROOT, 'package.json'));
+  }
+  return kibanaPackageFile;
+};
```

Comment:
If we want to avoid this, one option could be to allow this getter to be passed in at construction time, with a default implementation that uses a simple `require('../../../package.json')`.

From my understanding, this implementation only breaks in production builds, so we could replace this implementation to use REPO_ROOT in the main code path that starts up Kibana. This way tests continue to work as-is and production builds of Kibana will read the correct file.

Reply:
I don't really like having a default implementation that breaks in production tbh, but I don't see anything better.

I'm still not totally sure this approach will solve the current failures of tests mocking the `fs` module, as these tests are in `src/core` and/or `src/x-pack`, therefor using the `node_modules` version of this file, which may throw as the package.json is not at the specified place.

Overall, this package loading thing probably need to be totally reworked to be more solid